src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/AsType.cs

```diff
@@ -90,8 +90,9 @@ public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[
             // Check if table arg referrs to a connected data source.
             var tableArg = args[1];
             if (!binding.TryGetFirstNameInfo(tableArg.Id, out var tableInfo) ||
-                tableInfo.Data is not IExternalDataSource tableDsInfo ||
-                !(tableDsInfo is IExternalTabularDataSource))
+                (tableInfo.Data is not NameSymbol &&
```

Comment:
are there additional compile time checks we should do? Or do we punt to runtime?